### PR TITLE
When ghprb version is new enough, emit configVersion 3

### PR DESCRIPTION
**Pull request by mmitche**
Earlier versions of the ghprb did not have regex matching for trigger phrases.  When this was introduced, the config upgrade process quoted the trigger phrase strings.  Because the DSL does not emit configVersion 3, a regex trigger phrase will get quoted when it doesn't need to even to.

### CODE REVIEW

File: job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/helpers/triggers/TriggerContextSpec.groovy [hunk 1]
```diff
@@ -307,6 +307,35 @@ class TriggerContextSpec extends Specification {
...
+            triggerPhrase[0].value() == ''
+            adminlist[0].value() == ''
+            whitelist[0].value() == ''
+            orgslist[0].value() == ''
+            commentFilePath[0].value() == ''
+            configVersion[0].value() == '3'
+        }
+        1 * mockJobManagement.requirePlugin('ghprb')
+    }
+
```
Comment (daspilker) [inline job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/helpers/triggers/TriggerContextSpec.groovy]: remove the whitespace in this line to fix CodeNarc.

File: job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/helpers/triggers/TriggerContextSpec.groovy [hunk 2]
```diff
@@ -307,6 +307,35 @@ class TriggerContextSpec extends Specification {
...
+            commentFilePath[0].value() == ''
+            configVersion[0].value() == '3'
+        }
+        1 * mockJobManagement.requirePlugin('ghprb')
+    }
+    
+    def 'call pull request trigger with plugin version 1.26'() {
+        setup:
+        mockJobManagement.getPluginVersion('ghprb') >> new VersionNumber('1.26')
+
```
Comment (daspilker) [inline job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/helpers/triggers/TriggerContextSpec.groovy]: remove the whitespace in this line to fix CodeNarc.

**Comment by daspilker** [comment]
There may be some users who rely on this (mis)behavior. This change will break compatibility for them and the problem will probably be hard to spot.

Making the configVersion explicitly configurable would solve that. Everyone who needs the new behavior can set the version to 3 or whatever.

**Comment by mmitche** [comment]
@daspilker What about changing it to add a "regexTriggerPhrase" element?  If regexTriggerPhrase is used (mutually exclusive with triggerPhrase), then the output triggerPhrase is left unquoted. If triggerPhrase is used, the output triggerPhrase is wrapped with \Q \E (as the plugin would do on the upgrade path).  In both cases configVersion is set to 3 if the version is 1.27 or higher.

**Comment by mmitche** [comment]
Updated and rebased.

**Comment by daspilker** [comment]
But what if someone is using triggerPhrase with 1.26 or older? Then configVersion will not be set, but the triggerPhrase will be wrapped with \Q \E. That will break compatibility.

**Comment by daspilker** [comment]
I will contact the ghprb-plugin maintainer to see if we can move the DSL to the ghprb-plugin using the [extension point](https://github.com/jenkinsci/job-dsl-plugin/wiki/Extending-the-DSL). Then the DSL will always match the plugin's implementation and we don't have to track subtle changes.

**Comment by daspilker** [comment]
See [JENKINS-31214](https://issues.jenkins-ci.org/browse/JENKINS-31214).

**Comment by mmitche** [comment]
Ahh right sorry, should have a check there.

Sent from my Windows Phone
________________________________

**Comment by daspilker** [comment]
I got some positive feedback on JENKINS-31214. I will move the GitHub Pull Request Builder DSL to the ghprb plugin and deprecate pullRequest DSL method in the Job DSL plugin after the ghprb plugin has been released with DSL support.

This means that I will no longer accept changes for the pull request builder DSL. The DSL within ghprb plugin will then generate the latest config with version 3. I hope to get the changes done within the next two weeks. The ghprb plugin is released quite often, so this will be resolved soon.

**Comment by mmitche** [comment]
@daspilker That's great to know.  There are a couple other missing features (buildStatus comments) that I can help out implementing when the code moves to the plugin.

**Comment by daspilker** [comment]
Replaced by https://github.com/jenkinsci/job-dsl-plugin/pull/658.

**Pull request closed without merging**